clients/src/main/java/org/apache/kafka/common/requests/ApiError.java

```diff
@@ -36,10 +38,21 @@
     private final String message;
 
     public static ApiError fromThrowable(Throwable t) {
+        Throwable throwableToBeEncoded = t;
+        // Get the original cause from `CompletionException` and `ExecutionException`, otherwise, it will return unexpected UNKNOWN_SERVER_ERROR.
```

Comment:
nit: I think we can make this a little more concise. How about this?

```java
// Get the underlying cause for common exception types from the concurrent library. 
// This is useful to handle cases where exceptions may be raised from a future or a 
// completion stage (as might be the case for requests sent to the controller in 
// `ControllerApis`)
```

Reply:
Thanks for the suggestion! Updated.